Doc/whatsnew/3.13.rst

```diff
@@ -332,6 +332,14 @@ venv
   (using ``--without-scm-ignore-files``). (Contributed by Brett Cannon in
   :gh:`108125`.)
 
+warnings
+--------
+
+* The new :func:`warnings.deprecated` decorator provides a way to communicate
+  deprecations to :term:`static type checkers <static type checker>` and
+  to warn on usage of deprecated classes and functions.
```

Comment:
I still want to put the type checker effect  first because that's the unique part. You can write a decorator that generates runtime warnings yourself; the new and exciting part is that this decorator is also understood by static type checkers.

Reply:
You can do, sure, but _do_ people? It was possible to reimplement `itertools.batched` in a few lines of code before Python 3.12, but lots of people were still very excited by its inclusion in the stdlib in Python 3.12.

I think this new decorator here could prove pretty popular with people who don't use static typing! But, I don't feel strongly; it looks fine to me now :)

Reply:
Sure, there are lots of third-party deprecation decorators. The `Deprecated` library, I think Flask has one internally (David Lord mentioned it in PEP 702 discussions), I know at Quora we have one internally.